The nspawn container test framework seems to have been broken since staging-next was merged but once it is reverted the nixosTest is failing with the changes in this PR:

```
> bbmaster # [   14.478104] twistd[1169]: 2026-05-21T05:07:56+0000 [-] while starting BuildMaster
> bbmaster # [   14.479492] twistd[1169]:      Traceback (most recent call last):
> bbmaster # [   14.481106] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/buildbot/master.py", lie
> bbmaster # [   14.483834] twistd[1169]:          yield self.reconfigServiceWithBuildbotConfig(self.config)
> bbmaster # [   14.486273] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/twisted/internet/defer.s
> bbmaster # [   14.489049] twistd[1169]:          result = context.run(
> bbmaster # [   14.489986] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/twisted/python/failure.r
> bbmaster # [   14.493260] twistd[1169]:          return g.throw(self.value.with_traceback(self.tb))
> bbmaster # [   14.495108] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/buildbot/master.py", lig
> bbmaster # [   14.497899] twistd[1169]:          yield super().reconfigServiceWithBuildbotConfig(new_config)
> bbmaster # [   14.499892] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/twisted/internet/defer.s
> bbmaster # [   14.502978] twistd[1169]:          result = context.run(
> bbmaster # [   14.504276] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/twisted/python/failure.r
> bbmaster # [   14.507577] twistd[1169]:          return g.throw(self.value.with_traceback(self.tb))
> bbmaster # [   14.509213] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/buildbot/util/service.pg
> bbmaster # [   14.512945] twistd[1169]:          yield svc.reconfigServiceWithBuildbotConfig(new_config)
> bbmaster # [   14.514372] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/twisted/internet/defer.s
> bbmaster # [   14.517349] twistd[1169]:          result = context.run(gen.send, result)
> bbmaster # [   14.519116] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/buildbot/www/service.pyg
> bbmaster # [   14.521843] twistd[1169]:          self.setupSite(new_config)
> bbmaster # [   14.523115] twistd[1169]:        File "/nix/store/5g30zqs1w9409cds784b7c2k7mx972kj-python3-3.13.13-env/lib/python3.13/site-packages/buildbot/www/service.pye
> bbmaster # [   14.526338] twistd[1169]:          raise RuntimeError("could not find buildbot-www; is it installed?")
> bbmaster # [   14.528051] twistd[1169]:      builtins.RuntimeError: could not find buildbot-www; is it installed?
> bbmaster # [   14.529949] twistd[1169]:
```

--- a/pkgs/development/tools/continuous-integration/buildbot/master.nix
+++ b/pkgs/development/tools/continuous-integration/buildbot/master.nix
@@ -85,14 +85,13 @@ buildPythonApplication rec {
     hash = "sha256-yUtOJRI04/clCMImh5sokpj6MeBIXjEAdf9xnToqJZs=";
   };
 
-  build-system = [
-  ];
+  build-system = [ setuptools ];
 
   pythonRelaxDeps = [
     "twisted"
   ];
 
-  propagatedBuildInputs = [
+  dependencies = [
     # core
     twisted
     jinja2
@@ -105,7 +104,6 @@ buildPythonApplication rec {
     autobahn
     pyjwt
     pyyaml
-    setuptools
     croniter
     importlib-resources
     packaging

--- a/pkgs/development/tools/continuous-integration/buildbot/pkg.nix
+++ b/pkgs/development/tools/continuous-integration/buildbot/pkg.nix
@@ -1,14 +1,15 @@
 {
   lib,
   buildPythonPackage,
+  setuptools,
   isPy3k,
   buildbot,
 }:
 
 buildPythonPackage {
-  format = "setuptools";
   pname = "buildbot_pkg";
   inherit (buildbot) src version;
+  pyproject = true;
 
   postPatch = ''
     cd pkg
@@ -16,6 +17,8 @@ buildPythonPackage {
     # Do we have to care about that with Nix...?
     substituteInPlace buildbot_pkg.py --replace "os.listdir = listdir" ""
   '';
+
+  build-system = [ setuptools ];
 
   # No tests
   doCheck = false;

--- a/pkgs/development/tools/continuous-integration/buildbot/plugins.nix
+++ b/pkgs/development/tools/continuous-integration/buildbot/plugins.nix
@@ -2,7 +2,7 @@
   lib,
   buildPythonPackage,
   fetchurl,
-  callPackage,
+  setuptools,
   mock,
   cairosvg,
   klein,
@@ -13,9 +13,9 @@
   # this is exposed for potential plugins to use and for nix-update
   inherit buildbot-pkg;
   www = buildPythonPackage rec {
-    format = "setuptools";
     pname = "buildbot_www";
     inherit (buildbot-pkg) version;
+    pyproject = true;
 
     src = fetchurl {
       url = "https://github.com/buildbot/buildbot/releases/download/v${version}/${pname}-${version}.tar.gz";
@@ -27,7 +27,9 @@
       sed -i "s/'buildbot'//" setup.py
     '';
 
-    buildInputs = [
+    build-system = [ setuptools ];
+
+    dependencies = [
       buildbot-pkg
       mock
     ];
@@ -46,14 +48,16 @@
   console-view = buildPythonPackage rec {
     pname = "buildbot_console_view";
     inherit (buildbot-pkg) version;
-    format = "setuptools";
+    pyproject = true;
 
     src = fetchurl {
       url = "https://github.com/buildbot/buildbot/releases/download/v${version}/${pname}-${version}.tar.gz";
       hash = "sha256-VA6xqJBjD4XmQabTN8M+PLvfrG7Hq2ooxChtz2jAT8A=";
     };
 
-    buildInputs = [ buildbot-pkg ];
+    build-system = [ setuptools ];
+
+    dependencies = [ buildbot-pkg ];
 
     # No tests
     doCheck = false;
@@ -69,14 +73,16 @@
   waterfall-view = buildPythonPackage rec {
     pname = "buildbot_waterfall_view";
     inherit (buildbot-pkg) version;
-    format = "setuptools";
+    pyproject = true;
 
     src = fetchurl {
       url = "https://github.com/buildbot/buildbot/releases/download/v${version}/${pname}-${version}.tar.gz";
       hash = "sha256-c/Nmr0Uscalnndq72Y6jPM1JDs5OyOCERtuX/GXkxp8=";
     };
 
-    buildInputs = [ buildbot-pkg ];
+    build-system = [ setuptools ];
+
+    dependencies = [ buildbot-pkg ];
 
     # No tests
     doCheck = false;
@@ -92,14 +98,16 @@
   grid-view = buildPythonPackage rec {
     pname = "buildbot_grid_view";
     inherit (buildbot-pkg) version;
-    format = "setuptools";
+    pyproject = true;
 
     src = fetchurl {
       url = "https://github.com/buildbot/buildbot/releases/download/v${version}/${pname}-${version}.tar.gz";
       hash = "sha256-AmY8RkFX0POmVpW71nNz4+dFbr0FHGhNR3RJymDNoaw=";
     };
 
-    buildInputs = [ buildbot-pkg ];
+    build-system = [ setuptools ];
+
+    dependencies = [ buildbot-pkg ];
 
     # No tests
     doCheck = false;
@@ -115,14 +123,16 @@
   wsgi-dashboards = buildPythonPackage rec {
     pname = "buildbot_wsgi_dashboards";
     inherit (buildbot-pkg) version;
-    format = "setuptools";
+    pyproject = true;
 
     src = fetchurl {
       url = "https://github.com/buildbot/buildbot/releases/download/v${version}/${pname}-${version}.tar.gz";
       hash = "sha256-vofKxpIfbAs7HR43Y7ojHLQEn6/WIdjZPgZieBMsz74=";
     };
 
-    buildInputs = [ buildbot-pkg ];
+    build-system = [ setuptools ];
+
+    dependencies = [ buildbot-pkg ];
 
     # No tests
     doCheck = false;
@@ -138,15 +148,17 @@
   badges = buildPythonPackage rec {
     pname = "buildbot_badges";
     inherit (buildbot-pkg) version;
-    format = "setuptools";
+    pyproject = true;
 
     src = fetchurl {
       url = "https://github.com/buildbot/buildbot/releases/download/v${version}/${pname}-${version}.tar.gz";
       hash = "sha256-u7HF6X+ClT4rT3LJcTHXWi5oSxCKPXoUDH+QFRI2S0w=";
     };
 
-    buildInputs = [ buildbot-pkg ];
-    propagatedBuildInputs = [
+    build-system = [ setuptools ];
+
+    dependencies = [
+      buildbot-pkg
       cairosvg
       klein
       jinja2
@@ -163,5 +175,4 @@
       license = lib.licenses.gpl2;
     };
   };
-
 }

--- a/pkgs/development/tools/continuous-integration/buildbot/worker.nix
+++ b/pkgs/development/tools/continuous-integration/buildbot/worker.nix
@@ -7,6 +7,9 @@
   # patch
   coreutils,
 
+  # build system
+  setuptools,
+
   # propagates
   autobahn,
   msgpack,
@@ -15,7 +18,6 @@
   # tests
   parameterized,
   psutil,
-  setuptools-trial,
 
   # passthru
   nixosTests,
@@ -24,7 +26,7 @@
 buildPythonPackage {
   pname = "buildbot_worker";
   inherit (buildbot) src version;
-  format = "setuptools";
+  pyproject = true;
 
   postPatch = ''
     cd worker
@@ -33,11 +35,9 @@ buildPythonPackage {
       --replace /usr/bin/tail "${coreutils}/bin/tail"
   '';
 
-  nativeBuildInputs = [
-    setuptools-trial
-  ];
+  build-system = [ setuptools ];
 
-  propagatedBuildInputs = [
+  dependencies = [
     autobahn
     msgpack
     twisted

